src/test/tests.rs

```diff
@@ -60,33 +59,44 @@ struct ServerOptions {
 /// * An `Arc`-and-`Mutex`-wrapped `MockCommandCreator` which the server will
 ///   use for all process creation.
 /// * The `JoinHandle` for the server thread.
-fn run_server_thread<T: Into<Option<ServerOptions>> + Send + 'static>(cache_dir: &Path, options: T) -> (u16, Sender<ServerMessage>, Arc<Mutex<MockCommandCreator>>, thread::JoinHandle<()>) {
+fn run_server_thread<T>(cache_dir: &Path, options: T)
+                        -> (u16, Sender<ServerMessage>, Arc<Mutex<MockCommandCreator>>, thread::JoinHandle<()>)
+    where T: Into<Option<ServerOptions>> + Send + 'static
+{
     let options = options.into();
+    let cache_dir = cache_dir.to_path_buf();
+
     // Create a server on a background thread, get some useful bits from it.
     let (tx, rx) = mpsc::channel();
-    let storage = Box::new(DiskCache::new(&cache_dir, options.as_ref().and_then(|o| o.cache_size.as_ref()).map(|s| *s).unwrap_or(usize::MAX)));
+    let (shutdown_tx, shutdown_rx) = oneshot::channel();
     let handle = thread::spawn(move || {
-        let (mut server, event_loop) = create_server::<Arc<Mutex<MockCommandCreator>>>(0, storage).unwrap();
-        assert!(server.port() > 0);
+        let srv = SccacheServer::new(0).unwrap();
+        let mut srv: SccacheServer<Arc<Mutex<MockCommandCreator>>> = srv;
+        assert!(srv.port() > 0);
+        let cache_size = options.as_ref()
+                                .and_then(|o| o.cache_size.as_ref())
+                                .map(|s| *s)
+                                .unwrap_or(usize::MAX);
         if let Some(options) = options {
             if let Some(timeout) = options.idle_timeout {
-                 server.set_idle_timeout(timeout);
+                 srv.set_idle_timeout(Duration::from_millis(timeout));
             }
         }
-        let port = server.port();
-        let sender = event_loop.channel();
-        let creator = server.command_creator();
-        tx.send((port, sender, creator)).unwrap();
-        run_server(server, event_loop).unwrap()
+        let storage = Arc::new(DiskCache::new(&cache_dir, cache_size, srv.pool()));
+        srv.set_storage(storage.clone());
```

Comment:
Ah yeah the next PR queued up pushes futures all the way throughout caches and such, and I could totally plumb them that way. Right now the thread pool is passed into constructors but that could be easily changed I believe.